src/mux-analytics.brs

```diff
@@ -552,14 +563,28 @@ function muxAnalytics() as Object
     else
       if beacon.count() > 0
         m._logBeacon(beacon, "BEACON")
-       
+        retryCountdown% = m.HTTP_RETRIES
+        timeout% = m.HTTP_TIMEOUT
         m.connection.AsyncCancel()
         m.connection.SetUrl(m.beaconUrl)
         m.requestId = m.connection.GetIdentity()
         requestBody = {}
         requestBody.events = beacon
         fBody = FormatJson(requestBody)
-        success = m.connection.AsyncPostFromString(fBody)
+        while retryCountdown% > 0
+          m.connection.AsyncPostFromString(fBody)
+          event = wait(1500, m.httpPort)
```

Comment:
This doesn't _always_ wait 1500ms, does it? If so my suggestion above won't be a good thing (5 seconds instead of 1.5). If it is a static wait, is there no way to know asynchronously whether it succeeded or not?

Sorry, this brightscript is a little hard to understand what's going on without knowing Brightscript very well.

Reply:
@jsanford8 5sec is fine, it doesn't wait blindly until timeout. https://sdkdocs.roku.com/display/sdkdoc/Global+Utility+Functions#GlobalUtilityFunctions-Wait(timeoutasInteger,portasObject)asObject

@lounger should here use the m.httpTimeout instead of the hardcode number?

Reply:
Oh great, that makes sense, then. I'd honestly raise this to 10 seconds then, just to account for some really weird cases, as this is meant to just catch the case where we should give up and retry.

Reply:
I have made those changes. 
Yes the wait is an 'or' command. So it will wait for either the response or a timeout.